pandaserver/dataservice/adder_gen.py

```diff
@@ -117,411 +117,461 @@ def get_plugin_class(self, tmp_vo, tmp_group):
-                            self.taskBuffer.getEventRanges(self.job.PandaID, self.job.jobsetID, self.job.jediTaskID, 1, False, False, None)
-                        # check semaphore
-                        check_jc = self.taskBuffer.checkClonedJob(self.job)
-                        if check_jc is None:
-                            raise RuntimeError("failed to check the cloned job")
-                        # failed to lock semaphore
-                        if check_jc["lock"] is False:
-                            self.job_status = "failed"
-                            self.job.ddmErrorCode = pandaserver.dataservice.ErrorCode.EC_Adder
-                            self.job.ddmErrorDiag = "failed to lock semaphore for job cloning"
-                            self.logger.debug(f"set jobStatus={self.job_status} since did not get semaphore for job cloning")
-                # use failed for cancelled/closed jobs
-                if self.job.isCancelled():
-                    self.job_status = "failed"
-                    # reset error codes to skip retrial module
-                    self.job.pilotErrorCode = 0
-                    self.job.exeErrorCode = 0
-                    self.job.ddmErrorCode = 0
-                # keep old status
-                old_job_status = self.job.jobStatus
-                # set job status
-                if self.job.jobStatus not in ["transferring"]:
-                    self.job.jobStatus = self.job_status
-                add_result = None
-                adder_plugin = None
-                # parse JSON
-                parse_result = self.parse_json()
-                if parse_result < 2:
-                    # interaction with DDM
+                self.logger.debug("apply_retrial_rules is back")
+            except Exception as e:
+                self.logger.error(
+                    f"apply_retrial_rules excepted and needs to be investigated ({e}): {traceback.format_exc()}")
+
+        self.job.jobStatus = "failed"
+        for file in self.job.Files:
+            if file.type in ["output", "log"]:
+                if self.add_result is not None and file.lfn in self.add_result.merging_files:
+                    file.status = "merging"
+                else:
+                    file.status = "failed"
+
+    def check_job_status(self) -> None:
+        """
+        Check the job status and log appropriate messages.
+        """
+        if self.job is None:
+            self.logger.debug(": job not found in DB")
+        elif self.job.jobStatus in ["finished", "failed", "unknown", "merging"]:
+            self.logger.error(f": invalid state -> {self.job.jobStatus}")
+        elif self.attempt_nr is not None and self.job.attemptNr != self.attempt_nr:
+            self.logger.error(f"wrong attemptNr -> job={self.job.attemptNr} <> {self.attempt_nr}")
+        elif self.job_status == EventServiceUtils.esRegStatus:
+            self.register_event_service_files()
+        else:
```

Comment:
Since this function apparently does not return anything, why don't you add a return to the above if's and elif's. That would allow you to un-indent the gigantic else one level.